### PR TITLE
feature(plugins): moved hopeit.toolkit.storage.redis to a plugin

### PR DESCRIPTION
**Pull request by pcanto-hopeit**
MOTIVATION:
===========
Current version of hopeit.engine provides a toolkit for help easy way to access  Redis as key/value store.
Future versions of hopeit.engine will no more require `aioredis` library as dependency.

Plugins can be executed Standalone.

This plugin will be released as a python library and can be installed optionally when installing hopeit.engine with extras [redis-storage].

CHANGES ON THIS PR
===================
- [x] Move  `hopeit.toolkit.storage.redis` to a plugin: `hopeit.redis_storage`
- [x] Adapt CI to release `redis-storage` plugin to PyPi

CURRENT STATUS
===============
Ready for initial testing. Backwards compatible with previous implementation (*)

*Requires change on import:
1 - Install hopeit.engine using [redis-storage] plugin extras:

```
 pip install hopeit.redis-storage
```

2 - Change import signature

From:
```
from hopeit.toolkit.storage.redis import RedisStorage
```
To:
```
from hopeit.redis_storage import RedisStorage
```

### CODE REVIEW

**Comment by leosmerling** [comment]
Given that 0.2.4 and this both include small breaking chsnges, i th8nk it would be better to release as 0.3.0, so i suggest to set version to 0.3.0rc1 fir this pr.

We'll need also to port out toolkit.storage.fs to a plugin also in another PR.

**Comment by pcanto-hopeit** [comment]
> Given that 0.2.4 and this both include small breaking chsnges, i th8nk it would be better to release as 0.3.0, so i suggest to set version to 0.3.0rc1 fir this pr.
> 
> We'll need also to port out toolkit.storage.fs to a plugin also in another PR.

I agree to change the version, and already prepare the [PR](https://github.com/hopeit-git/hopeit.engine/pull/54) to replace toolkit.storage.fs.
It looks like 0.3.0rc1 is just around the corner

File: apps/benchmark/simple-benchmark/src/simple_benchmark/query_something_redis.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 from hopeit.app.api import event_api
 from hopeit.app.context import EventContext, PostprocessHook
 from hopeit.app.logger import app_extra_logger
-from hopeit.toolkit.storage.redis import RedisStorage
+from hopeit.redis_storage import RedisStorage
```
Comment (leosmerling-hopeit) [inline apps/benchmark/simple-benchmark/src/simple_benchmark/query_something_redis.py:13]: I don't see this added to requirements.txt/seutp.py of the benchmark.

File: .github/workflows/hopeit-engine-pypi-publishing.yml [hunk 2]
```diff
@@ -28,13 +28,22 @@ jobs:
...
+    - name: make plugin redis_storage
+      run: |
+        make PLUGINFOLDER=plugins/storage/redis/ dist-plugin
+    - name: Publish hopeit.engine on PyPI
       env: 
         PYPI_API_TOKEN: ${{ secrets.PYPI_API_TOKEN }}
       run: |
         make pypi
-    - name: Publish plugin redis_streams on Test PyPI
+    - name: Publish plugin redis_streams on PyPI
```
Comment (leosmerling-hopeit) [inline .github/workflows/hopeit-engine-pypi-publishing.yml:39]: Thank you!

File: plugins/storage/redis/src/hopeit/redis_storage/__init__.py [hunk 3]
```diff
@@ -8,8 +9,6 @@
 from hopeit.dataobjects import DataObject
 from hopeit.dataobjects.jsonify import Json
 
-__all__ = ['RedisStorage']
```
Comment (leosmerling-hopeit) [inline plugins/storage/redis/src/hopeit/redis_storage/__init__.py:11]: Why removed?

**Review by leosmerling-hopeit — COMMENTED** [review]
Only possible missing is requirements.txt/setup.py for benchmark with dependency on this plugin.

File: plugins/storage/redis/src/hopeit/redis_storage/__init__.py (hunk 3 above)
Comment (pcanto-hopeit) [reply plugins/storage/redis/src/hopeit/redis_storage/__init__.py]: Solved in 7d431ea

**Review by leosmerling-hopeit — CHANGES_REQUESTED** [review]
Add entry in release-notes about this change.

**Review by leosmerling-hopeit — APPROVED** [review]

**Pull request merged**